Source/WebCore/platform/graphics/avfoundation/objc/MediaPlayerPrivateMediaStreamAVFObjC.mm

```diff
@@ -750,19 +757,26 @@ void getSupportedTypes(HashSet<String>& types) const final
 
 void MediaPlayerPrivateMediaStreamAVFObjC::characteristicsChanged()
 {
-    bool sizeChanged = false;
+    SizeChanged sizeChanged = SizeChanged::No;
 
     FloatSize intrinsicSize = m_mediaStreamPrivate->intrinsicSize();
-    if (intrinsicSize.height() != m_intrinsicSize.height() || intrinsicSize.width() != m_intrinsicSize.width()) {
-        m_intrinsicSize = intrinsicSize;
-        sizeChanged = true;
-        if (m_playbackState == PlaybackState::None)
-            m_playbackState = PlaybackState::Paused;
+    if (intrinsicSize.isEmpty() || m_intrinsicSize.isEmpty()) {
+        if (intrinsicSize.height() != m_intrinsicSize.height() || intrinsicSize.width() != m_intrinsicSize.width()) {
+            m_intrinsicSize = intrinsicSize;
+            sizeChanged = SizeChanged::Yes;
+            if (m_playbackState == PlaybackState::None)
+                m_playbackState = PlaybackState::Paused;
```

Comment:
but does this indicate that it's paused?

There could be audio playing no?

Reply:
m_playbackState is used to compute the display mode.
This change allows to go from `render black` to `render paused image`.
Maybe we could improve the readability of the code here.